ethcore/src/types/transaction.rs

```diff
@@ -298,17 +285,9 @@ impl SignedTransaction {
 		s.append(&self.s);
 	}
 
-	/// Get the hash of this header (sha3 of the RLP).
-	pub fn hash(&self) -> H256 {
-		let hash = self.hash.get();
-		match hash {
-			Some(h) => h,
-			None => {
-				let h = (&*self.rlp_bytes()).sha3();
-				self.hash.set(Some(h));
-				h
-			}
-		}
+	/// Clones unsigned part of this transaction.
+	pub fn to_unsigned(&self) -> Transaction {
+		self.unsigned.clone()
```

Comment:
`as_unsigned -> &Transaction` with the relevant call sites using `as_unsigned().clone()` would fit the naming convention better.